compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/TextFieldPointerModifier.uikit.kt

```diff
@@ -0,0 +1,210 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.gestures.detectDragGesturesAfterLongPress
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.selection.SelectionAdjustment
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.foundation.text.selection.getTextFieldSelection
+import androidx.compose.foundation.text.selection.isSelectionHandleInVisibleBound
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.hapticfeedback.HapticFeedbackType
+import androidx.compose.ui.input.pointer.pointerHoverIcon
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TextFieldValue
+
+internal actual fun Modifier.textFieldPointer(
+    manager: TextFieldSelectionManager,
+    enabled: Boolean,
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping
+): Modifier = if (enabled) {
+    if (isInTouchMode) {
+        val selectionModifier = getSelectionModifier(manager)
+        val tapHandlerModifier = getTapHandlerModifier(
+            interactionSource,
+            state,
+            focusRequester,
+            readOnly,
+            offsetMapping,
+            manager
+        )
+        tapHandlerModifier
+            .then(selectionModifier)
+            .pointerHoverIcon(textPointerIcon)
+    } else {
+        Modifier
+            .mouseDragGestureDetector(
+                observer = manager.mouseSelectionObserver,
+                enabled = enabled
+            )
+            .pointerHoverIcon(textPointerIcon)
+    }
+} else {
+    Modifier
+}
+
+@OptIn(InternalFoundationTextApi::class)
+private fun getTapHandlerModifier(
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping,
+    manager: TextFieldSelectionManager
+) = Modifier.pointerInput(interactionSource) {
+    /*
+    We need to move tap recognizer here from selection modifier (as it is in common)
+    because otherwise we have onDoubleTap call and onTap call, and onDoubleTap will execute before onTap.
+     */
+    detectTapGestures(
+        onDoubleTap = { touchPointOffset ->
```

Comment:
For implementing triple clicks, see `mouseSelectionDetector` - it uses `clicksCounter` instead of `detectTapGestures`. We can refactor now, or in the triple click PR

Reply:
I see, this should work for double taps also, shouldn't it?
If so, I would rather refactor. Clicks counter logic seems to be a bit more lightweight.

Reply:
However, I'd rather do it when I'll start doing tripleTap handler. Too much refactoring here, I guess.

Reply:
onDoubleTap works here  just fine